lib/charms/ip_router_interface/v0/ip_router_interface.py

```diff
@@ -438,7 +440,7 @@ def request_network(self, networks: List[Network], custom_network_name: str = No
             raise RuntimeError("No ip-router relation exists yet.")
 
         for network_request in networks:
-            _validate_network(network_request, {"existing-networks": self.get_all_networks()})
```

Comment:
When I was locally testing the library, I thought it would be confusing for people to only see some of their networks in the databag when there was clearly an error raised. I could change the behavior if you think that should be the expected behavior.

Reply:
Me and @patriciareinoso talked about this, and decided to keep the existing behavior, and indicate that no routes were added in logs.